cou/cli.py

```diff
@@ -92,18 +96,29 @@ async def analyze_and_plan(model_name: Optional[str] = None) -> tuple[Analysis,
     :return: Generated analyses and upgrade plan.
     :rtype: tuple[Analysis, UpgradeStep]
     """
+    progress_indicator.start(f"Connecting to '{model_name or 'default'}' model...")
     model = await COUModel.create(model_name)
     logger.info("Using model: %s", model.name)
+    progress_indicator.succeed()
 
     progress_indicator.start("Analyzing cloud...")
     analysis_result = await Analysis.create(model)
-    progress_indicator.succeed()
     logger.info(analysis_result)
+    progress_indicator.succeed()
 
     progress_indicator.start("Generating upgrade plan...")
     upgrade_plan = await generate_plan(analysis_result)
     progress_indicator.succeed()
 
+    # NOTE(rgildein): add handling upgrade plan canceling for SIGINT (ctrl+c) and SIGTERM
+    loop = asyncio.get_event_loop()
+    loop.add_signal_handler(
+        SIGINT, keyboard_interrupt_handler, upgrade_plan, loop, progress_indicator
+    )
+    loop.add_signal_handler(
+        SIGTERM, keyboard_interrupt_handler, upgrade_plan, loop, progress_indicator
+    )
```

Comment:
It seems that the safe cancellation handling only works with executing `UpgradeStep`. If that's the case, wouldn't it make more sense to move these lines to `run_upgrade` function?

Also, perhaps we want to also handle KeyboardInterrupt safely for other async tasks (e.g. `COUModel.create(model_name)`, `generate_plan(analysis_result)`). What do you think?